pyccel/parser/semantic.py

```diff
@@ -2071,6 +2071,15 @@ def _visit(self, expr):
             severity='fatal')
 
     def _visit_Module(self, expr):
+        for f in expr.funcs:
+            self.insert_function(f)
+
+        for i in expr.imports:
+            self._visit(i)
+
+        for c in expr.classes:
+            self._visit(c)
```

Comment:
I agree but I suspect this is already the case. It is very similar to the problem with functions that you fixed. For the moment it seems classes in functions are not supported as they are not mentioned in the arguments
https://github.com/pyccel/pyccel/blob/b39fe9a411736a833d980674b7f6380278febf0f/pyccel/ast/core.py#L2228-L2298

Reply:
Actually won't these problems only arise inside class methods? In which case it should also be handled by your fix?

Reply:
Ah, no these are annotated directly:
https://github.com/pyccel/pyccel/blob/b39fe9a411736a833d980674b7f6380278febf0f/pyccel/parser/semantic.py#L4347-L4348

Reply:
https://github.com/pyccel/pyccel/issues/1814

Reply:
As this is already broken I would rather leave these changes for a second PR

Reply:
ah yes, that is another issue, but I was talking about something like
```python
a = 1
class A:
    x = a
```
This would have worked before but not anymore

Reply:
ah yes, that is another issue, but I was talking about something like
```python
a = 1
class A:
    x = a
```
This would have worked before but not anymore

Reply:
I agree but given that the fix is the same for both I don't know if it is worth fixing here? I can't just visit the classes later otherwise the more common case of:
```python
class A:
    pass

my_a = A()
```
will break

Reply:
Or we can put this PR on standby, fix #1814 and then come back to clean it up in this PR

Reply:
As you prefer, we just need to make sure that we handle those cases

Reply:
The issue should ensure we don't forget